pkg/sources/github/github.go

```diff
@@ -450,23 +453,23 @@ func (s *Source) ensureRepoInfoCache(ctx context.Context, repo string) (string,
 		for {
 			gistID := extractGistID(urlParts)
 			gist, _, err := s.connector.APIClient().Gists.Get(ctx, gistID)
-			// Normalize the URL to the Gist's pull URL.
-			// See https://github.com/trufflesecurity/trufflehog/pull/2625#issuecomment-2025507937
-			repo = gist.GetGitPullURL()
-			if s.handleRateLimit(ctx, err) {
+			if s.handleRateLimit(ctx, err, reporter) {
 				continue
 			}
 			if err != nil {
 				return repo, fmt.Errorf("failed to fetch gist")
 			}
+			// Normalize the URL to the Gist's pull URL.
+			// See https://github.com/trufflesecurity/trufflehog/pull/2625#issuecomment-2025507937
+			repo = gist.GetGitPullURL()
```

Comment:
Why was this added here? It looks unrelated to error reporting.

Reply:
it was just moved down. we don't need that URL until after we handled the rate limit. although on a second look, it should probably be before the err check, because we return the repo